packages/eslint-plugin/src/rules/no-useless-template-expression.ts

```diff
@@ -0,0 +1,194 @@
+
+                // Remove the quasis and backticks.
+                return [
+                  fixer.removeRange([
+                    prevQuasi.range[1] - 3,
+                    node.expressions[0].range[0],
+                  ]),
+
+                  fixer.removeRange([
+                    node.expressions[0].range[1],
+                    nextQuasi.range[0] + 2,
+                  ]),
+                ];
+              },
+            });
+
+            return;
+          }
+
+          const fixableExpressions = node.expressions.filter(
+            expression =>
+              isLiteral(expression) ||
+              isTemplateLiteral(expression) ||
+              isUndefinedIdentifier(expression) ||
+              isInfinityIdentifier(expression) ||
+              isNaNIdentifier(expression),
+          );
+
+          fixableExpressions.forEach(expression => {
+            context.report({
+              node: expression,
+              messageId: 'noUselessTemplateExpression',
+              fix(fixer): TSESLint.RuleFix[] {
+                const index = node.expressions.indexOf(expression);
+                const prevQuasi = node.quasis[index];
+                const nextQuasi = node.quasis[index + 1];
+
+                // Remove the quasis' parts that are related to the current expression.
+                const fixes = [
+                  fixer.removeRange([
+                    prevQuasi.range[1] - 2,
+                    expression.range[0],
+                  ]),
+
+                  fixer.removeRange([
+                    expression.range[1],
+                    nextQuasi.range[0] + 1,
+                  ]),
+                ];
+
+                const stringValue = getStaticStringValue(expression);
+
+                if (stringValue != null) {
+                  const escapedValue = stringValue.replace(/([`$\\])/g, '\\$1');
+
+                  fixes.push(fixer.replaceText(expression, escapedValue));
+                } else if (isTemplateLiteral(expression)) {
+                  // Note that some template literals get handled in the previous branch too.
+                  // Remove the beginning and trailing backtick characters.
+                  fixes.push(
+                    fixer.removeRange([
+                      expression.range[0],
+                      expression.range[0] + 1,
+                    ]),
+                    fixer.removeRange([
+                      expression.range[1] - 1,
+                      expression.range[1],
+                    ]),
+                  );
+                }
+
+                return fixes;
+              },
+            });
+          });
+        },
+      };
+    },
+  };
+
+  return ruleObject;
+}
+
+// The rule object factory is to support code sharing between the current rule,
+// and the deprecated version of this rule under its previous name
+// no-useless-template-literals.
+// See https://github.com/typescript-eslint/typescript-eslint/issues/8544
```

Comment:
[Docs] So this gets picked up in searches for todos...

```suggestion
// See https://github.com/typescript-eslint/typescript-eslint/issues/8544
TODO: Remove in v8
```

...which reminds me, I should note this in #8913. We have a [mention of todo searches for ESLint versions](https://github.com/typescript-eslint/typescript-eslint/blob/edb3f68beec5e13bdafc453afedc6d31f7ecfe06/docs/maintenance/pull-requests/Dependency_Version_Upgrades.mdx#L88) but not for our majors.

https://github.com/typescript-eslint/typescript-eslint/pull/8913/commits/f262b165ec58c3b85a32ca307bcfa8bee3a96522